core/cap-0062.md

```diff
@@ -0,0 +1,281 @@
+term gains from "partial state archival" introduced in this CAP. Specifically, this CAP will allow for full in-memory Soroban live state
+caching and automatic restoration of Soroban entries via `InvokeHostFunctionOp`, to be detailed in a future CAP.
+
+### Fee Changes
+
+Because ledger state is now split into two BucketLists, the BucketList size component for storage fees must be
+reconsidered. Currently, the total size of the BucketList is used for Soroban related fees. This is frustrating
+to end users, as classic state dominates the BucketList size such that changes unrelated to Soroban have the most
+impact towards Soroban related fees. The purpose of the fee is to give back pressure to new writes when the BucketList
+is large. This should discourage writes (or financially prevent DOS attacks) and give time for State Archival to evict
+state, reduce the size of the BucketList, thus reducing fees. The issue is, classic entries dominate storage and are
+not subject to State Archival, so this back pressure system is not effective. In practice, should BucketList size
+rapidly increase, it is most likely due to classic state, and the only way to lower fees is via network config upgrade.
+
+This CAP changes the BucketList size used in fee calculations to Live Soroban State Size. Specifically, Live Soroban
+State Size is is the total amount of state currently consumed by Soroban entries in the Live BucketList, in bytes.
+This includes TTL entries and shadows.
+
+The Hot Archive size does not impact fees. Long term, it is expected that CAP-0057 will be implemented and the
+Hot Archive will be dropped from validators. Thus long term, from a network health standpoint, the size of
+the Hot Archive does not have a significant impact on network sustainability.
+
+Prior to CAP-0057, large Hot Archives do impact network sustainability, specifically with respect to History
+Archive size, catchup time, and node disk requirements. However, a fee based on Hot Archive size does not
+seem to address this issue. First, there is currently no way to remove state from the Hot Archive. Only
+restore operations remove Hot Archive state, but this state is just added back to the Live BucketList. Thus
+fees would continually increase without any benefit to network sustainability, as no action or system is
+currently in place to reduce Hot Archive size.
+
+While not improving network sustainability, Hot Archive fees could prevent state based DOS attacks, where
+state is rapidly added to the network. However, unlike the live BucketList, operations cannot directly
+write state to the Hot Archive (with the exception of restores, which always decrease Hot Archive size).
+The rate of eviction (i.e. the rate at which state is added to the Hot Archive) can already be controlled
+via network config settings. The rate at which state is added to the Live BucketList is also already
```

Comment:
Can you elaborate on this? I was under the impression that eviction is controlled by state TTL rather than network configurations. Or is this referencing the eviction of already-archived entries in the live BL?

Reply:
Ya this is the eviction of already archived entries. The Live BucketList at any given time may hold an arbitrary number of archived entries (since the TTL alone dictates this). However the rate at which we evict these archived entries (remove them from the Live BucketList and add them to the Hot Archive BucketList) is restricted via network config settings (specifically there are settings for both max entries to evict and max bytes to evict).

Reply:
Note that this is an implementation detail, the restore Op looks the same to the invoker whether or not the archived entry has been evicted or not.

Reply:
Right yep, understood that and hence wanted to clarify that that's exactly what was being discussed here